### ISSUE TITLE
Anonymous function in struct breaks indentation

### ISSUE THREAD

**Issue by andy-hanson** [dlang-community/dfmt#432]
Input:
```d
import std.stdio : writeln;

struct S {
	ulong x;
	ulong y;
	ulong function(ulong) f;
}

immutable S s = {
	1111111111111111111,
	1111111111111111111,
	(x) {
		return x + 1111;
	},
};

void main() {
	writeln(s.f(1));
}
```

Output:
```d
import std.stdio : writeln;

struct S
{
    ulong x;
    ulong y;
    ulong function(ulong) f;
}

immutable S s = {
    1111111111111111111, 1111111111111111111, (x) { return x + 1111; },};

    void main()
    {
        writeln(s.f(1));
    }
```

**Comment by andy-hanson** [comment]
It looks like an anonymous function and a long struct initializer are needed, but there doesn't need to be nesting:
```d
import std.stdio : writeln;

struct S {
	ulong x;
	ulong y;
	ulong z;
	ulong w;
}

immutable int function(int) f = (x) {
	return x + 1111;
};

immutable S s = {
	1111111111111111111,
	1111111111111111111,
	1111111111111111111,
	1111111111111111111,
};

void main() {
}
```
becomes:
```d
struct S
{
    ulong x;
    ulong y;
    ulong z;
    ulong w;
}

immutable int function(int) f = (x) { return x + 1111; };

immutable S s = {
    1111111111111111111, 1111111111111111111, 1111111111111111111, 1111111111111111111,};

    void main()
    {
    }
```

**Comment by WebFreak001** [comment]
fixed in #590
